vcd/resource_vcd_org_vdc.go

```diff
@@ -0,0 +1,502 @@
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdOrgVdc() *schema.Resource {
+	capacityWithUsage := schema.Schema{
+		Type:     schema.TypeSet,
+		Required: true,
+		ForceNew: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"units": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ForceNew:     true,
+					ValidateFunc: validation.StringInSlice([]string{"MHz", "GHz", "MB", "GB"}, false),
+					Description:  "Units in which capacity is allocated. For CPU capacity, one of: {MHz, GHz}.  For memory capacity, one of: {MB, GB}.",
+				},
+				"allocated": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Capacity that is committed to be available.",
+				},
+				"limit": {
+					Type:        schema.TypeInt,
+					Required:    true,
+					ForceNew:    true,
+					Description: "Capacity limit relative to the value specified for Allocation. It must not be less than that value. If it is greater than that value, it implies over provisioning.",
+				},
+				"reserved": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Capacity reserved",
+				},
+				"used": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Capacity used. If the VDC AllocationModel is ReservationPool, this number represents the percentage of the reservation that is in use. For all other allocation models, it represents the percentage of the allocation that is in use.",
+				},
+				"overhead": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Number of Units allocated to system resources such as vShield Manager virtual machines and shadow virtual machines provisioned from this Provider VDC.",
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceVcdVdcCreate,
+		Delete: resourceVcdVdcDelete,
+		Read:   resourceVcdVdcRead,
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Organization to create the VDC in",
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"allocation_model": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"AllocationVApp", "AllocationPool", "ReservationPool"}, false),
+				Description:  "The allocation model used by this VDC; must be one of {AllocationVApp, AllocationPool, ReservationPool}",
+			},
+			"compute_capacity": &schema.Schema{
+				Required: true,
+				ForceNew: true,
```

Comment:
Worth adding for better validation and avoiding index out of range panics.
```
MaxItems     1,
MinItems      1
```

Reply:
Added. And yet we have inside check too for that

Reply:
Yeah, but that should allow `terraform validate`. Great!